src/shared/comparisons.ts

```diff
@@ -57,18 +51,38 @@ export const declaredInitializedTypeNodeIsRedundant = (
 		return undefined;
 	}
 
-	return declaredTypeIsEquivalent(
-		request.services.program.getTypeChecker(),
-		declaredType,
-		initializedType,
-	);
+	const typeChecker = request.services.program.getTypeChecker();
+
+	const declaredText = declaredType.getSymbol()?.getEscapedName();
+
+	// This is brute-force way to keep Map<string, number> comparison to Map without type arguments...
+	// This will allow many type declarations that could be removed.
+	if (
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+		(declaredText === "Map" ||
```

Comment:
I feel very icky about hardcoding to Maps & Sets by name 😬. Those two classes just so happen to be declared in `lib.es*.d.ts`. Users might declare their own classes that we'd want to have the same kind of fixing behavior on. Same with interfaces like `Readonlymap`.

Reply:
Yeah I think it sets a bad precedent. I have actually seen a bunch of codebases use real names like `Map` and `Set`. And the older the codebase, the more likely it wants a tool like TypeStat to help it migrate!